src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -19,56 +20,57 @@ namespace Microsoft.Extensions.Caching.StackExchangeRedis;
 /// </summary>
 public partial class RedisCache : IDistributedCache, IDisposable
 {
-    // -- Explanation of why two kinds of SetScript are used --
-    // * Redis 2.0 had HSET key field value for setting individual hash fields,
-    // and HMSET key field value [field value ...] for setting multiple hash fields (against the same key).
-    // * Redis 4.0 added HSET key field value [field value ...] and deprecated HMSET.
+    // * Key timeouts are used in the following way:
     //
-    // On Redis versions that don't have the newer HSET variant, we use SetScriptPreExtendedSetCommand
-    // which uses the (now deprecated) HMSET.
-
-    // KEYS[1] = = key
-    // ARGV[1] = absolute-expiration - ticks as long (-1 for none)
-    // ARGV[2] = sliding-expiration - ticks as long (-1 for none)
-    // ARGV[3] = relative-expiration (long, in seconds, -1 for none) - Min(absolute-expiration - Now, sliding-expiration)
-    // ARGV[4] = data - byte[]
-    // this order should not change LUA script depends on it
-    private const string SetScript = (@"
-                redis.call('HSET', KEYS[1], 'absexp', ARGV[1], 'sldexp', ARGV[2], 'data', ARGV[4])
-                if ARGV[3] ~= '-1' then
-                  redis.call('EXPIRE', KEYS[1], ARGV[3])
-                end
-                return 1");
-    private const string SetScriptPreExtendedSetCommand = (@"
-                redis.call('HMSET', KEYS[1], 'absexp', ARGV[1], 'sldexp', ARGV[2], 'data', ARGV[4])
-                if ARGV[3] ~= '-1' then
-                  redis.call('EXPIRE', KEYS[1], ARGV[3])
-                end
-                return 1");
-
-    private const string AbsoluteExpirationKey = "absexp";
-    private const string SlidingExpirationKey = "sldexp";
-    private const string DataKey = "data";
-    private const long NotPresent = -1;
-    private static readonly Version ServerVersionWithExtendedSetCommand = new Version(4, 0, 0);
+    // * If an entry has no absolute expiration time or sliding window, then
+    //   the key in Redis has no timeout value.
+    //
+    // * If an entry has only an absolute expiration time, then the key in Redis
+    //   has a corresponding timeout.
+    //
+    // * If an entry has a sliding window, then the key in Redis has a timeout value
+    //   corresponding to the sliding window. Every fetch of that entry resets the
+    //   key's timeout to (now + sliding window). This is potentially limited by an
+    //   absolute expiration time, which sets an upper bound.
+
+    private readonly Func<string, string> _expandKey;
+    private readonly RedisCacheOptions _options;
+    private readonly ILogger _logger;
```

Comment:
Not used anymore. Do we want to log other things though (more things)?

Reply:
How do we log things?

Reply:
Should we add logs for hit/miss?
What about metrics?

Reply:
Hit/miss would be too noisy, that would have to be Trace level.